File: java/kata/src/test/java/br/com/ari/algo/structures/QueueTest.java
package br.com.ari.algo.structures;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class QueueTest {

    @Test
    void testQueue() {
        Queue<Integer> queue = new Queue<>();
        queue.enqueue(1);
        queue.enqueue(2);
        queue.enqueue(3);
        queue.enqueue(4);
        queue.enqueue(5);

        assertThat(queue.size()).isEqualTo(5);
        assertThat(queue.peek()).isEqualTo(1);
        assertThat(queue.dequeue()).isEqualTo(1);
        assertThat(queue.peek()).isEqualTo(2);
        assertThat(queue.size()).isEqualTo(4);

        int[] expectedValues = new int[] { 2, 3, 4, 5 };
        for (int expected : expectedValues) {
            assertThat(queue.dequeue()).isEqualTo(expected);
        }

        assertThat(queue.size()).isZero();
    }

}

File: go/kata/structures/queue_test.go
package structures

import (
	"github.com/stretchr/testify/assert"
	"testing"
)

func TestQueue(t *testing.T) {
	queue := NewQueue[int]()
	assert.Equal(t, 0, queue.Peek())
	assert.Equal(t, 0, queue.Dequeue())

	queue.Enqueue(1)
	queue.Enqueue(2)
	queue.Enqueue(3)
	queue.Enqueue(4)
	queue.Enqueue(5)

	assert.Equal(t, 1, queue.Peek())
	assert.Equal(t, 1, queue.Peek())
	assert.Equal(t, 1, queue.Peek())

	assert.Equal(t, 1, queue.Dequeue())
	assert.Equal(t, 2, queue.Dequeue())
	assert.Equal(t, 3, queue.Dequeue())
	assert.Equal(t, 4, queue.Dequeue())
	assert.Equal(t, 5, queue.Dequeue())
	assert.Equal(t, 0, queue.Dequeue())
	assert.Equal(t, 0, queue.Dequeue())
	assert.Equal(t, 0, queue.Dequeue())

	queue.Enqueue(5)
	queue.Enqueue(4)
	queue.Enqueue(3)
	queue.Enqueue(2)
	queue.Enqueue(1)

	assert.Equal(t, 5, queue.Dequeue())
	assert.Equal(t, 4, queue.Dequeue())
	assert.Equal(t, 3, queue.Dequeue())
	assert.Equal(t, 2, queue.Dequeue())
	assert.Equal(t, 1, queue.Dequeue())
	assert.Equal(t, 0, queue.Dequeue())
}
